### PR TITLE
Split out pure-Django feature flags from Wagtail-flags

### PR DESCRIPTION
**Pull request by willbarton**
This PR removes references to Wagtail-Flags and makes the library independent of Wagtail so that it can be used with plain Django, addressing https://github.com/cfpb/wagtail-flags/issues/28.

### CODE REVIEW

**Comment by willbarton** [comment]
This PR also moves the documentation from a huge single-page README into readthedocs style documentation: http://cfpb.github.io/django-flags

**Comment by willbarton** [comment]
@higs4281 Thanks for the review of the docs! I'll add your language to the intro page and finding more wagtails to purge. 

I think the "Edit on GitHub" link looks like it'll be correct when the docs are on master.

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+Then add `flags` as an installed app in your Django `settings.py`:
+
+```python
+INSTALLED_APPS = (
+    ...
+    'flags',
+    ...
+)
+```
+
```
[the remark this answers is not in the document]
Comment (willbarton) [reply docs/index.md]: Done!

File: docs/index.md [hunk 2]
```diff
@@ -19,9 +19,19 @@ Then add `flags` as an installed app in your Django `settings.py`:
...
     'flags',
-    ...
+    # …
+)
+```
+
+And `django.template.context_processors.request` to the `TEMPLATE_CONTEXT_PROCESSORS` setting:
+
+```python
+TEMPLATE_CONTEXT_PROCESSORS = (
```
Comment (higs4281) [inline docs/index.md]: `TEMPLATE_CONTEXT_PROCESSORS` is deprecated.

Shouldn't we use the `TEMPLATES` settings style?

Comment (willbarton) [reply docs/index.md]: 🙈

**Pull request merged**